### PR TITLE
FAT-2668/C350650

### PR DESCRIPTION
**Pull request by re-jas**
[Implement e-2-e automation of test case C350650: Verify automated patron block 'Maximum number of overdue recalls' removed after overdue recalled item returned](https://issues.folio.org/browse/FAT-2668)

### CODE REVIEW

File: cypress/support/fragments/settings/users/limits.js [hunk 1]
```diff
@@ -44,4 +45,14 @@ export default {
...
+  setLimit: (limitName, number) => {
+    Conditions.getConditionsViaApi().then(conditions => {
+      return conditions.find(condition => condition.name === limitName).id;
+    }).then((limitId) => {
+      cy.do([
+        TextField({ name: limitId }).fillIn(number),
+        saveButton.click(),
+      ]);
+    });
+  }
```
Comment (re-jas) [inline cypress/support/fragments/settings/users/limits.js:57]: Can I delete other setMaximumNumber... methods and instead use this one in other tests where the limit needs to be specified? In my opinion, this method is better and more versatile.

Comment (penfriend) [reply cypress/support/fragments/settings/users/limits.js:57]: Sure! Great approach!

File: cypress/support/fragments/settings/users/limits.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import uuid from 'uuid';
+import Conditions from './conditions';
 import { NavListItem, TextField, Button } from '../../../../../interactors';
 
 const saveButton = Button('Save');
```
Comment (penfriend) [inline cypress/support/fragments/settings/users/limits.js:5]: please update function name on the 8th line to "createViaApi"

File: cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+          }).then((specialInstanceIds) => {
+            itemsData.itemsWithSeparateInstance[index].instanceId = specialInstanceIds.instanceId;
+            itemsData.itemsWithSeparateInstance[index].holdingId = specialInstanceIds.holdingIds[0].id;
+            itemsData.itemsWithSeparateInstance[index].itemId = specialInstanceIds.holdingIds[0].itemIds;
+          });
+        });
+        cy.wrap(itemsData.itemsWithSeparateInstance).as('items');
+      });
+
+    LoanPolicy.createApi(loanPolicyBody);
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js:122]: please rename to LoanPolicy.createViaApi(loanPolicyBody);

Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js:122]: Done

File: cypress/support/fragments/settings/users/limits.js (hunk 2 above)
Comment (re-jas) [reply cypress/support/fragments/settings/users/limits.js:5]: done

File: cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+      CirculationRules.addRuleViaApi(originalCirculationRules, ruleProps, 't ', testData.loanTypeId);
+    });
+
+    cy.createTempUser(
+      [
+        permissions.uiUsersSettingsOwners.gui,
+        permissions.uiUsersCreatePatronConditions.gui,
+        permissions.uiUsersCreatePatronLimits.gui,
+        permissions.checkinAll.gui,
+        permissions.checkoutAll.gui,
```
[the remark this answers is not in the document]
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js:139]: removed

File: cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+        servicePointId: testData.userServicePoint.id,
+        checkInDate: new Date().toISOString(),
+      });
+    });
+    cy.deleteLoanPolicy(loanPolicyBody.id);
+    UserEdit.changeServicePointPreferenceViaApi(userData.userId, [testData.userServicePoint.id]);
+    UserEdit.changeServicePointPreferenceViaApi(recallUserData.userId, [testData.userServicePoint.id]);
+    ServicePoints.deleteViaApi(testData.userServicePoint.id);
+    testData.requestsId.forEach((id) => {
+      Requests.deleteRequestApi(id);
```
[the remark this answers is not in the document]
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js:207]: done

File: cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+        itemBarcode: item.barcode,
+        servicePointId: testData.userServicePoint.id,
+        checkInDate: new Date().toISOString(),
+      });
+    });
+    cy.deleteLoanPolicy(loanPolicyBody.id);
+    UserEdit.changeServicePointPreferenceViaApi(userData.userId, [testData.userServicePoint.id]);
+    UserEdit.changeServicePointPreferenceViaApi(recallUserData.userId, [testData.userServicePoint.id]);
+    ServicePoints.deleteViaApi(testData.userServicePoint.id);
+    testData.requestsId.forEach((id) => {
```
[the remark this answers is not in the document]
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/patron-block-maximum-number-overdue-recalls-removed-after-overdue-recalled-item-returned.cy.js:206]: done

File: cypress/e2e/settings/users/patron-block/patron-block-maximum-items-overdue-removed-after-overdue-item-returned.cy.js [hunk 7]
```diff
@@ -153,7 +153,7 @@ describe('Patron Block: Maximum number of overdue items', () => {
         permissions.uiUsersCreatePatronConditions.gui,
         permissions.uiUsersCreatePatronLimits.gui,
         permissions.checkinAll.gui,
-        permissions.checkoutAll.gui,
+        // permissions.checkoutAll.gui,
```
Comment (IhorBohdan) [inline cypress/e2e/settings/users/patron-block/patron-block-maximum-items-overdue-removed-after-overdue-item-returned.cy.js:156]: Need here this comment?

**Review by IhorBohdan — APPROVED** [review]

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/e2e/settings/users/patron-block/patron-block-maximum-items-overdue-removed-after-overdue-item-returned.cy.js (hunk 7 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/patron-block-maximum-items-overdue-removed-after-overdue-item-returned.cy.js]: fixed

**Pull request merged**